ckan/controllers/group.py

```diff
@@ -104,22 +99,25 @@ def _render_template(self, template_name):
         ''' render the correct group/org template '''
         return render(self._replace_group_org(template_name))
 
-    def _redirect_to(self, *args, **kw):
-        ''' wrapper to ensue the correct controller is used '''
-        if self.group_type == 'organization' and 'controller' in kw:
-            kw['controller'] = 'organization'
+    def _redirect_to__this_controller(self, *args, **kw):
+        ''' wrapper around redirect_to but it adds in this request's controller
+        (so that it works for Organization or other derived controllers)'''
+        kw['controller'] = request.environ['pylons.routes_dict']['controller']
```

Comment:
Mild significance... just a separator, like two spaces. This method is a variation on the familiar func 'redirect_to'. Rather than reading 'redirect to this controller', it more means 'redirect to whatever the *args/**kw say, as long as it is in this controller'. I'll remove the two if you prefer or can think of a better name.